src/api_manager/context/request_context.cc

```diff
@@ -376,15 +384,30 @@ const std::string RequestContext::FindClientIPAddress() {
 void RequestContext::StartBackendSpanAndSetTraceContext() {
   backend_span_.reset(CreateSpan(cloud_trace_.get(), "Backend"));
 
-  // The span id in the header will be the backend span's id.
-  std::string trace_context = cloud_trace()->ToTraceContextHeader(
-      backend_span_->trace_span()->span_id());
+  // TODO: A better logic would be to send for GRPC backends the grpc-trace-bin
+  // header, and for http/https backends the X-Cloud-Trace-Context header.
+
+  std::string trace_context_header;
+  if (cloud_trace()->header_type() == HeaderType::GRPC_TRACE_CONTEXT) {
+    // The span id in the header will be the backend span's id.
+    std::string base64_trace_context_header =
```

Comment:
this name is confusing.  It is raw_,  not base64 encoded.

Reply:
Actually is not raw, the base64 comes from grpc/esp integration. If I use directly the grpc::ClientContext I will but the non-base64 (raw) value, but the gRPC protocol does base64 encoding and you keep that inside esp (in the proxy code for example you explicitly base64 encode all the binary headers).

Reply:
Done, renamed and moved.